src/tcp_sock.rs

```diff
@@ -358,3 +316,94 @@ impl Drop for Inner {
         let _ = self.stream.shutdown(Shutdown::Both);
     }
 }
+
+/// Serialize given value and write to a buffer with 4 byte length header.
+fn write_len_delimited<T: Serialize>(value: T) -> ::Res<Vec<u8>> {
```

Comment:
I borrowed the naming from tokio: https://docs.rs/tokio/0.1.12/tokio/codec/length_delimited/struct.LengthDelimitedCodec.html
I thought it makes sense because of two things:
1. frames in TCP stream are actually delimited by length, e.g. len1, data1, len2, data2, len3, data3..
    Also, by definition `delimit - To establish the limits or boundaries of`this makes sense because data is "bounded by it's length".
2. such naming establishes a common terminology between socket-collection and tokio project and sort of follow the conventions used in other communities. So potentially could save some cognitive resources for newcomers :)

Reply:
> This is not writing anything (in protocol/TCP sense here).

well, it writes to an in-memory buffer. Regarding sockets `send()` is used instead (`man 2 send`).